lib/Wikia/src/Service/User/Attributes/AttributeKeyValueService.php

```diff
@@ -87,16 +87,29 @@ public function delete( $userId, Attribute $attribute ) {
 			$ret = $this->persistenceAdapter->deleteAttribute( $userId, $attribute );
 			return $ret;
 		} catch ( \Exception $e ) {
-			$this->logError( $userId, $e, "USER_ATTRIBUTES error deleting from service" );
+			$this->logException( $userId, $e, "USER_ATTRIBUTES failure deleting from service" );
 			return false;
 		}
 	}
 
-	private function logError( $userId, \Exception $e, $msg ) {
-		$this->error( $msg , [
-			'user' => $userId,
-			'exception' => $e
-		] );
+	/**
+	 * Log any exceptions thrown when contacting the attribute service. There are 4 possible exceptions
+	 * which can be thrown, and which this method can expect to receive: PersistenceException, ForbiddenException,
+	 * NotFoundException, and UnauthorizedException. Log PersistenceExceptions at error level, and all others
+	 * at info. PersistenceExceptions indicate a problem was encountered contacting the service, or that we received
+	 * an unexpected HTTP response like a 500. The other exceptions indicate a resources wasn't found, or that the
+	 * request was refused which is not an error.
+	 * @param $userId
+	 * @param \Exception $e
+	 * @param $msg
+	 */
+	private function logException( $userId, \Exception $e, $msg ) {
+		$context = [ 'user' => $userId, 'exception' => $e ];
+		if ( is_a( $e, 'Wikia\Service\PersistenceException' ) ) {
```

Comment:
No real reason. I googled "check if object is type of class php" and the `is_a` function is the first thing that came up. It looks like `intanceof` is a better approach though. I'll update.